### PR TITLE
Fix linux build and add build.sh

### PR DESCRIPTION
**Pull request by ta264**
This fixes the build on linux (xbuild complains about a few ambiguous calls) and adds build.sh, the first item in #506 

_output_mono seems to work for me.  _output_osx and _output_osx_app look ok compared to running build.ps1 on windows but I don't have a mac to test it on.  _output is missing all the pdbs - it might be possible to generate them but I'm not sure if there's any need.
http://stackoverflow.com/questions/13499384/is-it-possible-to-debug-assemblies-compiled-with-mono-xbuild-with-visual-studi

_tests looks ok and mono_tests.sh runs with almost all the tests passing.  However the integration tests  from NzbDrone.Integration.Test.dll all seem to fail.  I've put a log file [here](https://gist.github.com/ta264/e37f0db2e5b49014e02b) in case you can see what's going wrong.  I tried running them in VS from the delevop branch and they still fail, with a similar file not found error, so I don't think it's anything I've done.

### CODE REVIEW

**Comment by ta264** [comment]
Just noticed that aaff2965454e58c46fdb84149092a64eeb728587 breaks the windows build...

**Comment by ta264** [comment]
I think this should work on windows and linux now.

**Comment by ta264** [comment]
The integration tests are fixed with #663

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,179 @@
+#! /bin/bash
```
Comment (kayone) [inline build.sh]: I would love to replace our powershell script with this, much rather maintain a single platform agnostic one vs having two different ones for windows and *nix.

Comment (kayone) [inline build.sh]: I'm gussing you used the `build.ps` as base and just translated it over?

File: src/NzbDrone.Api/Authentication/EnableAuthInNancy.cs [hunk 2]
```diff
@@ -39,8 +39,8 @@ public void Register(IPipelines pipelines)
                 pipelines.EnableBasicAuthentication(new BasicAuthenticationConfiguration(_authenticationService, "Sonarr"));                
             }
 
-            pipelines.BeforeRequest.AddItemToEndOfPipeline(RequiresAuthentication);
-            pipelines.AfterRequest.AddItemToEndOfPipeline(RemoveLoginHooksForApiCalls);
+            pipelines.BeforeRequest.AddItemToEndOfPipeline((Func<NancyContext, Response>) RequiresAuthentication);
```
Comment (kayone) [inline src/NzbDrone.Api/Authentication/EnableAuthInNancy.cs]: I think someone did this before and it broke something in production, @markus101 knows the details.

Comment (kayone) [inline src/NzbDrone.Api/Authentication/EnableAuthInNancy.cs]: https://github.com/Sonarr/Sonarr/pull/202#issuecomment-75893962

Comment (markus101) [inline src/NzbDrone.Api/Authentication/EnableAuthInNancy.cs]: I believe this is safe (and @ta264 already realized the other way was broken), running the PR build to confirm though.

File: build.sh (hunk 1 above)
Comment (ta264) [inline build.sh]: Yup, just translated it function by function

Comment (ta264) [inline build.sh]: I had a look and I think I should be able to make this work for the windows build, using git bash which should have been installed with git. Is that what you had in mind?

Comment (kayone) [inline build.sh]: yes, we already have a dependency on git, and the build script is not really needed for day to day dev, it mostly just runs on the build servers.

**Comment by ta264** [comment]
I've updated it so that it works on windows too.  As far as I can tell it gives the same output as running build.ps1 on windows.

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+CleanFolder()
+{
+    local path=$1
+    local keepConfigFiles=$2
+
+    echo "Removing XMLDoc files"
+    find $path -name "*.xml" -delete
+    find $path -name "*.transform" -delete
```
Comment (kayone) [inline build.sh]: I'm getting an error here in windows, 

```
Removing XMLDoc files
find: invalid predicate `-delete'
find: invalid predicate `-delete'
find: invalid predicate `-delete'
Removing FluentValidation.Resources files
find: invalid predicate `-delete'
find: invalid predicate `-delete'
Removing Empty folders
find: invalid predicate `-delete'
```

Comment (ta264) [inline build.sh]: Maybe that was only added in the new msys with git2... I can switch it to
use -exec rm

Comment (kayone) [inline build.sh]: let me update my msys.

Comment (ta264) [inline build.sh]: I'm using RC 5 from here https://github.com/git-for-windows/git/releases

Comment (kayone) [inline build.sh]: update to latest version, still same issue.

Comment (kayone) [inline build.sh]: I think it would be nice if we could keep it compatible with latest stable, some people might not want to run pre-release software.

Comment (markus101) [inline build.sh]: The `-delete` flag on find isn't supported by all systems, most do, but I don't think we can expect it to work.

Comment (ta264) [inline build.sh]: I've swapped it to use exec rm. I'm getting some warnings when it deletes
empty directories - I'll have a look tomorrow

Comment (ta264) [inline build.sh]: I think i've fixed the problem with CleanFolder now.  I also double checked
the files produced and, for me at least, I get identical output when I use
this or build.ps1 on windows.  The output on linux is slightly different -
a few mdbs for the NuGet packages are missing and it adds an extra
dll, Interop.NetFwTypeLib.dll

**Comment by kayone** [comment]
@ta264 can you squash this so we can merge it in,

Thanks.

**Comment by TomAndrews** [comment]
@kayone I've pushed a squashed commit.  Thanks.

**Comment by kayone** [comment]
I tried running the build in OSX and got, 

```
/Users/kayone/git/Sonarr/src/NzbDrone.sln (Build) ->
(Build target) ->
/Users/kayone/git/Sonarr/src/NzbDrone.Common/NzbDrone.Common.csproj (default targets) ->
/Library/Frameworks/Mono.framework/Versions/4.0.2/lib/mono/4.5/Microsoft.CSharp.targets (CoreCompile target) ->

	Cache/Cached.cs(19,17): error CS0118: `object' is a `type' but a `variable' was expected
	Cache/Cached.cs(19,17): error CS0131: The left-hand side of an assignment must be a variable, a property or an indexer
	Composition/ContainerBuilderBase.cs(27,13): error CS0118: `NzbDrone.Common.Composition.Container' is a `type' but a `variable' was expected
	Composition/ContainerBuilderBase.cs(27,29): error CS0131: The left-hand side of an assignment must be a variable, a property or an indexer
	Exceptron/ExceptronClient.cs(71,13): error CS0118: `NzbDrone.Common.Exceptron.Configuration' is a `namespace' but a `variable' was expected
	Exceptron/ExceptronClient.cs(71,13): error CS0131: The left-hand side of an assignment must be a variable, a property or an indexer
	Http/HttpRequest.cs(14,13): error CS0118: `System.UriBuilder' is a `type' but a `variable' was expected
	Http/HttpRequest.cs(14,30): error CS0131: The left-hand side of an assignment must be a variable, a property or an indexer
	TinyIoC.cs(3007,17): error CS0118: `System.Type' is a `type' but a `variable' was expected
	TinyIoC.cs(3005,37): error CS0131: The left-hand side of an assignment must be a variable, a property or an indexer

```

**Comment by kayone** [comment]
merged

**Pull request closed without merging**